demos/common/python/openvino/model_zoo/model_api/models/blur.py

```diff
@@ -0,0 +1,74 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import cv2
+import numpy as np
+from typing import Any, Dict, Optional
+
+from .segmentation import SegmentationModel
+from .types import NumericalValue
+from ..adapters.model_adapter import ModelAdapter
+from .utils import check_input_parameters_type, create_hard_prediction_from_soft_prediction
+
+
+class BlurSegmentation(SegmentationModel):
+    __model__ = 'blur_segmentation'
+
+    @check_input_parameters_type()
+    def __init__(self, model_adapter: ModelAdapter, configuration: Optional[dict] = None, preload: bool = False):
+        super().__init__(model_adapter, configuration, preload)
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'soft_threshold': NumericalValue(default_value=0.5, min=0.0, max=1.0),
+            'blur_strength': NumericalValue(value_type=int, default_value=1, min=0, max=25)
+        })
+
+        return parameters
+
+    def _check_io_number(self, number_of_inputs, number_of_outputs):
```

Comment:
why do you ignore number of inputs/outputs?
Does it mean that it can be any?

Reply:
hmm... this class inherited from our Segmentation model wrapper, where preprocessing step expects that model have 1 inputs. Probably you can just call basic method with parameters
(1, -1). -1 is specific value which speak that number of outputs should be ignored

Reply:
Thank you for your comment! I'll also check this part and investigate if there is side effect for this :)